.circleci/config.yml

```diff
@@ -0,0 +1,125 @@
+version: 2.1
+
+executors:
+  ruby:
+    parameters:
+      version:
+        description: "Ruby version number"
+        default: "3.1.0"
+        type: string
+    docker:
+      - image: cimg/ruby:<< parameters.version >>
```

Comment:
This is the preferred docker image to use for Ruby builds. It only has tags for recent Ruby versions, though, so we have to use a "legacy" image for testing older versions of Ruby.